lib/Expr/SourceBuilder.cpp

```diff
@@ -21,7 +22,7 @@ SourceBuilder::uninitialized(unsigned version, const KInstruction *allocSite) {
 }
 
 ref<SymbolicSource> SourceBuilder::symbolicSizeConstantAddress(
-    unsigned version, const KInstruction *allocSite, ref<Expr> size) {
+    unsigned version, const KValue *allocSite, ref<Expr> size) {
```

Comment:
We need some way to restrict `allocSite` to those types of `KValue` that make sense as alloc sites. At least make a comment somewhere describing accepted derivatives or make a set of constructors accepting those derivatives only.